<!--
%\VignetteEngine{knitr::docco_classic}
%\VignetteIndexEntry{bold vignette}
-->

`bold` is an R package to connect to BOLD Systems \url{http://www.boldsystems.org/} via their API. Functions in `bold` let you search for sequence data, specimen data, sequence + specimen data, and download raw trace files.

### bold info

+ [BOLD home page](http://boldsystems.org/)
+ [BOLD API docs](boldsystems.org/index.php/resources/api)

### Using bold

**Install**

Install `bold` from CRAN



```r
install.packages("bold")
```

Or install the development version from GitHub


```r
devtools::install_github("ropensci/bold")
```

Load the package


```r
library("bold")
```

### Search for sequence data only

The BOLD sequence API gives back sequence data, with a bit of metadata.

The default is to get a list back


```r
bold_seq(taxon='Coelioxys')[1:2]
```

```
## [[1]]
## [[1]]$id
## [1] "BBHYL404-10"
## 
## [[1]]$name
## [1] "Coelioxys rufitarsis"
## 
## [[1]]$gene
## [1] "BBHYL404-10"
## 
## [[1]]$sequence
## [1] "TATAATATATATAATTTTTGCAATATGATCAGGTATAATTGGATCATCTTTAAGAATAATTATTCGAATAGAATTAAGAATCCCAGGTTCATGAATTAGAAATGATCAAATTTATAATTCTTTTATTACAGCACATGCATTTTTAATAATTTTTTTTTTAGTTATGCCTTTTCTAATTGGGGGATTTGGTAATTGATTAACACCATTAATACTTGGAGCTCCTGATATAGCTTTCCCCCGAATAAACAATATTAGATTTTGACTACTCCCACCTTCTTTATTACTTTTATTATCAAGAAATTTAATTAATCCAAGACCAGGAACAGGATGAACTGTTTATCCACCATTATCCTCTTATACATATCATCCATCTCCTTCTGTAGATTTAGCAATTTTTTCTTTACATTTATCAGGAATTTCCTCAATTATTGGATCAATAAATTTTATTGTTACAATTTTAATAATAAAAAATTATTCAATAAATTATAATCAAATACCATTATTCCCATGATCAGTTTTAATTACTACAATTTTATTATTACTATCACTTCCAGTATTAGCAGGAGCAATTACAATATTATTATTTGATCGAAATTTAAATTCTTCTTTTTTTGACCCAATAGGAGGAGGAGACCCAATTTTATATCAACATTTATTT"
## 
## 
## [[2]]
## [[2]]$id
## [1] "CNEID3374-12"
## 
## [[2]]$name
## [1] "Coelioxys funeraria"
## 
## [[2]]$gene
## [1] "CNEID3374-12"
## 
## [[2]]$sequence
## [1] "TATAATATATATAATTTTTGCAATATGATCAGGAATAATTGGTTCTTCATTAAGAATAATTATCCGAATAGAATTAAGAATCCCAGGATCTTGAATTAATAATGATCAAATTTATAATTCTTTTATCACAGCTCACGCATTTTTAATAATTTTTTTTTTAGTTATACCTTTTTTAATTGGAGGATTTGGTAATTGATTAGCCCCTTTAATATTAGGAGCCCCTGATATAGCTTTCCCTCGAATAAACAACATTAGATTTTGATTATTACCTCCTTCCTTATTAATACTTTTATCTAGTAATTTAATTACCCCTAGACCAGGGACAGGGTGAACTATTTACCCTCCATTATCTTTATATAATTATCANCCTTCTCCATCTGTTGATTTAGCAATTTTTTCTTTACATTTATCAGGAATTTCCTCTATTATTGGATCAATAAATTTCATTGTAACAATTTTAATAATAAAAAATTTTTCAATAAATTACAATCAAATACCTTTATTTCCATGATCAATTCTAATTACTACAATTTTATTATTATTATCATTACCTGTATTAGCAGGAGCAATTACAATATTATTATTTGATCGTAATTTAAATTCTTCATTTTTTGATCCAATAGGAGGAGGAGA---------"
```

You can optionally get back the `httr` response object


```r
res <- bold_seq(taxon='Coelioxys', response=TRUE)
res$headers
```

```
## $date
## [1] "Thu, 24 Jul 2014 19:03:24 GMT"
## 
## $server
## [1] "Apache/2.2.15 (Red Hat)"
## 
## $`x-powered-by`
## [1] "PHP/5.3.15"
## 
## $`content-disposition`
## [1] "attachment; filename=fasta.fas"
## 
## $connection
## [1] "close"
## 
## $`transfer-encoding`
## [1] "chunked"
## 
## $`content-type`
## [1] "application/x-download"
## 
## attr(,"class")
## [1] "insensitive" "list"
```

You can do geographic searches


```r
bold_seq(geo = "USA")
```

```
## [[1]]
## [[1]]$id
## [1] "NEONV108-11"
## 
## [[1]]$name
## [1] "Aedes thelcter"
## 
## [[1]]$gene
## [1] "NEONV108-11"
## 
## [[1]]$sequence
## [1] "AACTTTATACTTCATCTTCGGAGTTTGATCAGGAATAGTTGGTACATCATTAAGAATTTTAATTCGTGCTGAATTAAGTCAACCAGGTATATTTATTGGAAATGACCAAATTTATAATGTAATTGTTACAGCTCATGCTTTTATTATAATTTTCTTTATAGTTATACCTATTATAATTGGAGGATTTGGAAATTGACTAGTTCCTCTAATATTAGGAGCCCCAGATATAGCTTTCCCTCGAATAAATAATATAAGTTTTTGAATACTACCTCCCTCATTAACTCTTCTACTTTCAAGTAGTATAGTAGAAAATGGATCAGGAACAGGATGAACAGTTTATCCACCTCTTTCATCTGGAACTGCTCATGCAGGAGCCTCTGTTGATTTAACTATTTTTTCTCTTCATTTAGCCGGAGTTTCATCAATTTTAGGGGCTGTAAATTTTATTACTACTGTAATTAATATACGATCTGCAGGAATTACTCTTGATCGACTACCTTTATTCGTTTGATCTGTAGTAATTACAGCTGTTTTATTACTTCTTTCACTTCCTGTATTAGCTGGAGCTATTACAATACTATTAACTGATCGAAATTTAAATACATCTTTCTTTGATCCAATTGGAGGAGGAGACCCAATTTTATACCAACATTTATTT"
## 
## 
## [[2]]
## [[2]]$id
## [1] "NEONV109-11"
## 
## [[2]]$name
## [1] "Aedes thelcter"
## 
## [[2]]$gene
## [1] "NEONV109-11"
## 
## [[2]]$sequence
## [1] "AACTTTATACTTCATCTTCGGAGTTTGATCAGGAATAGTTGGTACATCATTAAGAATTTTAATTCGTGCTGAATTAAGTCAACCAGGTATATTTATTGGAAATGACCAAATTTATAATGTAATTGTTACAGCTCATGCTTTTATTATAATTTTCTTTATAGTTATACCTATTATAATTGGAGGATTTGGAAATTGACTAGTTCCTCTAATATTAGGAGCCCCAGATATAGCTTTCCCTCGAATAAATAATATAAGTTTTTGAATACTACCTCCCTCATTAACTCTTCTACTTTCAAGTAGTATAGTAGAAAATGGGTCAGGAACAGGATGAACAGTTTATCCACCTCTTTCATCTGGAACTGCTCATGCAGGAGCCTCTGTTGATTTAACTATTTTTTCTCTTCATTTAGCCGGAGTTTCATCAATTTTAGGGGCTGTAAATTTTATTACTACTGTAATTAATATACGATCTGCAGGAATTACTCTTGATCGACTACCTTTATTCGTTTGATCTGTAGTAATTACAGCTGTTTTATTACTTCTTTCACTTCCTGTATTAGCTGGAGCTATTACAATACTATTAACTGATCGAAATTTAAATACATCTTTCTTTGACCCAATTGGAGGGGGAGACCCAATTTTATACCAACATTTATTT"
```

And you can search by researcher name


```r
bold_seq(researchers='Thibaud Decaens')[[1]]
```

```
## $id
## [1] "AMAZ116-09"
## 
## $name
## [1] "Citheronia phoronea"
## 
## $gene
## [1] "AMAZ116-09"
## 
## $sequence
## [1] "AACTTTATATTTTATTTTTGGAATTTGAGCAGGAATAGTTGGAACTTCATTAAGATTACTAATTCGAGCTGAATTAGGTACCCCCGGATCTTTAATTGGAGATGATCAAATTTACAATACTATTGTTACAGCACATGCTTTCATTATAATTTTTTTTATGGTTATACCTATTATAATTGGAGGATTTGGTAATTGATTAGTACCCTTAATATTAGGAGCACCAGATATAGCTTTCCCCCGAATAAATAATATAAGTTTTTGATTATTACCCCCATCCTTAACTCTTCTAATTTCCAGAAGAATTGTAGAAAATGGAGCAGGAACAGGATGAACAGTATACCCCCCTCTCTCTTCTAATATTGCTCATGGAGGATCTTCAGTAGATTTAGCAATTTTTTCCCTTCATTTGGCAGGAATTTCTTCAATTTTAGGAGCTATTAACTTTATTACTACAATTATTAATATACGATTAAATAATATATCATTTGATCAAATACCCCTATTTGTTTGAGCTGTCGGAATTACAGCTTTCCTTTTATTATTATCTTTACCTGTATTAGCAGGAGCTATTACCATACTTCTAACTGATCGAAATTTAAATACTTCTTTTTTTGATCCTGCAGGAGGAGGAGATCCTATTTTATATCAACATTTATTC"
```

by taxon IDs


```r
bold_seq(ids=c('ACRJP618-11','ACRJP619-11'))
```

```
## [[1]]
## [[1]]$id
## [1] "ACRJP618-11"
## 
## [[1]]$name
## [1] "Lepidoptera"
## 
## [[1]]$gene
## [1] "ACRJP618-11"
## 
## [[1]]$sequence
## [1] "------------------------TTGAGCAGGCATAGTAGGAACTTCTCTTAGTCTTATTATTCGAACAGAATTAGGAAATCCAGGATTTTTAATTGGAGATGATCAAATCTACAATACTATTGTTACGGCTCATGCTTTTATTATAATTTTTTTTATAGTTATACCTATTATAATTGGAGGATTTGGTAATTGATTAGTTCCCCTTATACTAGGAGCCCCAGATATAGCTTTCCCTCGAATAAACAATATAAGTTTTTGGCTTCTTCCCCCTTCACTATTACTTTTAATTTCCAGAAGAATTGTTGAAAATGGAGCTGGAACTGGATGAACAGTTTATCCCCCACTGTCATCTAATATTGCCCATAGAGGTACATCAGTAGATTTAGCTATTTTTTCTTTACATTTAGCAGGTATTTCCTCTATTTTAGGAGCGATTAATTTTATTACTACAATTATTAATATACGAATTAACAGTATAAATTATGATCAAATACCACTATTTGTGTGATCAGTAGGAATTACTGCTTTACTCTTATTACTTTCTCTTCCAGTATTAGCAGGTGCTATCACTATATTATTAACGGATCGAAATTTAAATACATCATTTTTTGATCCTGCAGGAGGAGGAGATCCAATTTTATATCAACATTTATTT"
## 
## 
## [[2]]
## [[2]]$id
## [1] "ACRJP619-11"
## 
## [[2]]$name
## [1] "Lepidoptera"
## 
## [[2]]$gene
## [1] "ACRJP619-11"
## 
## [[2]]$sequence
## [1] "AACTTTATATTTTATTTTTGGTATTTGAGCAGGCATAGTAGGAACTTCTCTTAGTCTTATTATTCGAACAGAATTAGGAAATCCAGGATTTTTAATTGGAGATGATCAAATCTACAATACTATTGTTACGGCTCATGCTTTTATTATAATTTTTTTTATAGTTATACCTATTATAATTGGAGGATTTGGTAATTGATTAGTTCCCCTTATACTAGGAGCCCCAGATATAGCTTTCCCTCGAATAAACAATATAAGTTTTTGGCTTCTTCCCCCTTCACTATTACTTTTAATTTCCAGAAGAATTGTTGAAAATGGAGCTGGAACTGGATGAACAGTTTATCCCCCACTGTCATCTAATATTGCCCATAGAGGTACATCAGTAGATTTAGCTATTTTTTCTTTACATTTAGCAGGTATTTCCTCTATTTTAGGAGCGATTAATTTTATTACTACAATTATTAATATACGAATTAACAGTATAAATTATGATCAAATACCACTATTTGTGTGATCAGTAGGAATTACTGCTTTACTCTTATTACTTTCTCTTCCAGTATTAGCAGGTGCTATCACTATATTATTAACGGATCGAAATTTAAATACATCATTTTTTGATCCTGCAGGAGGAGGAGATCCAATTTTATATCAACATTTATTT"
```

by container (containers include project codes and dataset codes)


```r
bold_seq(container='ACRJP')[[1]]
```

```
## $id
## [1] "ACRJP142-09"
## 
## $name
## [1] "Lepidoptera"
## 
## $gene
## [1] "ACRJP142-09"
## 
## $sequence
## [1] "AACTTTATATTTTATCTTTGGTATCTGATCTGGGATAATTGGTACTTCTTTAAGTTTACTAATTCGTACAGAATTAGGAAATCCAGGCTCTTTAATTGGAGATGATCAAATTTATAATACTATTGTAACAGCACATGCTTTTATTATAATTTTCTTTATAGTTATACCAATTATAATTGGAGGCTTCGGAAATTGATTAGTTCCATTAATATTAGGAGCCCCTGATATAGCTTTCCCCCGAATAAATAATATAAGTTTTTGACTTCTTCCCCCTTCATTAATTTTATTAGTTTCTAGAAGAATTGTAGAAAATGGAGCAGGAACCGGATGAACAGTCTATCCCCCTCTTTCATCAAATATTGCTCATGGAGGCTCCTCAGTAGATTTAGCTATTTTCTCTCTTCATTTAGCCGGAATTTCCTCTATTCTAGGAGCAATTAATTTTATCACAACAATTATTAATATACGAATTAATAATTTATCCTTTGATCAAATACCCCTATTCGTTTGAGCAGTAGGAATTACAGCTCTTCTACTTCTTTTATCTTTACCAGTATTAGCAGGAGCTATTACTATACTTTTAACAGACCGAAATTTAAATACTTCTTTTTTTGACCCTGCAGGAGGAGGAGACCCAATTTTATACCAACACTTATTT"
```

by bin (a bin is a _Barcode Index Number_)


```r
bold_seq(bin='BOLD:AAA5125')[[1]]
```

```
## $id
## [1] "BLPAB406-06"
## 
## $name
## [1] "Eacles ormondei"
## 
## $gene
## [1] "BLPAB406-06"
## 
## $sequence
## [1] "AACTTTATATTTTATTTTTGGAATTTGAGCAGGTATAGTAGGAACTTCTTTAAGATTACTAATTCGAGCAGAATTAGGTACCCCCGGATCTTTAATTGGAGATGACCAAATTTATAATACCATTGTAACAGCTCATGCTTTTATTATAATTTTTTTTATAGTTATACCTATTATAATTGGAGGATTTGGAAATTGATTAGTACCCCTAATACTAGGAGCTCCTGATATAGCTTTCCCCCGAATAAATAATATAAGATTTTGACTATTACCCCCATCTTTAACTCTTTTAATTTCTAGAAGAATTGTCGAAAATGGAGCTGGAACTGGATGAACAGTTTATCCCCCCCTTTCATCTAATATTGCTCATGGAGGCTCTTCTGTTGATTTAGCTATTTTTTCCCTTCATCTAGCTGGAATCTCATCAATTTTAGGAGCTATTAATTTTATCACAACAATCATTAATATACGACTAAATAATATAATATTTGACCAAATACCTTTATTTGTATGAGCTGTTGGTATTACAGCATTTCTTTTATTGTTATCTTTACCTGTACTAGCTGGAGCTATTACTATACTTTTAACAGATCGAAACTTAAATACATCATTTTTTGACCCAGCAGGAGGAGGAGATCCTATTCTCTATCAACATTTATTT"
```

And there are more ways to query, check out the docs for `?bold_seq`.


### Search for specimen data only

The BOLD specimen API doesn't give back sequences, only specimen data. By default you download `tsv` format data, which is given back to you as a `data.frame`


```r
res <- bold_specimens(taxon='Osmia')
head(res[,1:8])
```

```
##     processid      sampleid recordID    catalognum       fieldnum
## 1 BBHYL359-10  10BBCHY-3313  1769802  10BBCHY-3313 L#PC2010YO-001
## 2 CHUBE002-06 CHU05-BEE-002   516711 CHU05-BEE-002  CHU05-BEE-002
## 3 GBAH0293-06      AF250940   470890                             
## 4 GBAH3909-08      EU726598   856440      EU726598               
## 5 GBAH3911-08      EU726596   856442      EU726596               
## 6 GBAH3937-08      EU726570   856468      EU726570               
##                                            institution_storing
## 1                            Biodiversity Institute of Ontario
## 2 University of Manitoba, Wallis Roughley Museum of Entomology
## 3                                     Mined from GenBank, NCBI
## 4                                     Mined from GenBank, NCBI
## 5                                     Mined from GenBank, NCBI
## 6                                     Mined from GenBank, NCBI
##        bin_uri phylum_taxID
## 1 BOLD:AAF2159           20
## 2 BOLD:AAD4181           20
## 3 BOLD:AAD6282           20
## 4 BOLD:AAA4494           20
## 5 BOLD:AAA4494           20
## 6 BOLD:AAA4494           20
```

You can optionally get back the data in `XML` format


```r
bold_specimens(taxon='Osmia', format='xml')
```


```r
<?xml version="1.0" encoding="UTF-8"?>
<bold_records  xsi:noNamespaceSchemaLocation="http://www.boldsystems.org/schemas/BOLDPublic_record.xsd"  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <record>
    <record_id>1470124</record_id>
    <processid>BOM1525-10</processid>
    <bin_uri>BOLD:AAN3337</bin_uri>
    <specimen_identifiers>
      <sampleid>DHB 1011</sampleid>
      <catalognum>DHB 1011</catalognum>
      <fieldnum>DHB1011</fieldnum>
      <institution_storing>Marjorie Barrick Museum</institution_storing>
    </specimen_identifiers>
    <taxonomy>
```

You can choose to get the `httr` response object back if you'd rather work with the raw data returned from the BOLD API.


```r
res <- bold_specimens(taxon='Osmia', format='xml', response=TRUE)
res$url
```

```
## [1] "http://www.boldsystems.org/index.php/API_Public/specimen?taxon=Osmia&specimen_download=xml"
```

```r
res$status_code
```

```
## [1] 200
```

```r
res$headers
```

```
## $date
## [1] "Thu, 24 Jul 2014 19:10:02 GMT"
## 
## $server
## [1] "Apache/2.2.15 (Red Hat)"
## 
## $`x-powered-by`
## [1] "PHP/5.3.15"
## 
## $`content-disposition`
## [1] "attachment; filename=bold_data.xml"
## 
## $connection
## [1] "close"
## 
## $`transfer-encoding`
## [1] "chunked"
## 
## $`content-type`
## [1] "application/x-download"
## 
## attr(,"class")
## [1] "insensitive" "list"
```

### Search for specimen plus sequence data

The specimen/sequence combined API gives back specimen and sequence data. Like the specimen API, this one gives by default `tsv` format data, which is given back to you as a `data.frame`. Here, we're setting `sepfasta=TRUE` so that the sequence data is given back as a list, and taken out of the `data.frame` returned so the `data.frame` is more manageable.


```r
res <- bold_seqspec(taxon='Osmia', sepfasta=TRUE)
res$fasta[1:2]
```

```
## $`BBHYL359-10`
## [1] "AATTTTATATATAATTTTTGCTATATGAGCAGGTATAATTGGATCAGCAATAAGAATTATTATTCGTATAGAATTAAGAATTCCTGGTTCATGAATCTCAAATGATCAAACTTATAATTCTTTAGTAACTGCTCATGCCTTTTTAATAATTTTTTTTCTAGTTATACCATTTCTAATTGGAGGATTTGGAAATTGATTAATTCCTTTAATATTAGGAATTCCAGATATAGCTTTTCCACGAATAAATAATATTAGATTTTGACTTTTACCACCTTCTTTAATATTATTACTCCTAAGAAATTTTCTTAATCCAAGTCCAGGAACTGGATGAACTGTCTATCCACCTCTCTCTTCTCATTTATTTCATTCTTCCCCTTCAGTTGATATAGCTATTTTTTCTTTACATATTTCCGGTTTATCTTCTATTATAGGCTCATTAAATTTTATTGTTACAATTATTATAATAAAAAATATTTCTTTAAAACATATTCAATTACCTTTATTTCCCTGATCTGTTTTTATTACTACTATTTTACTTCTTTTTTCTTTACCTGTTTTAGCAGGTGCTATTACTATATTATTATTTGATCGAAATTTTAATACTTCATTTTTTGATCCTACAGGAGGAGGAGATCCAATTCTTTATCAACATTTATTT"
## 
## $`CHUBE002-06`
## [1] "-----TATATATAATTTTTGCTATATGATCAGGAATAATTGGTTCAGCAATAAGAATTATTATTCGTATAGAATTAAGAATTCCTGGTTCATGAATTTCAAATGATCAAATTTATAATTCTTTAGTAACTGCTCATGCTTTTTTAATAATTTTTTTTTTAGTAATACCATTTTTAATTGGGGGATTTGGAAATTGATTAATTCCTTTAATATTAGGAATTCCAGATATAGCATTTCCACGAATAAATAATATTAGATTTTGACTTTTACCTCCTTCTTTAATACTTTTACTTTTAAGAAATTTTCTTAATCCAAGTCCAGGAACTGGATGAACTGTTTATCCTCCTCTTTCTTCACATTTATTTCATTCTTCGCCTTCAGTTGATATAGCTATTTTTTCCTTACATATTTCTGGTTTATCTTCTATTATAGGTTCATTAAATTTTATTGTAACAATTATTATAATAAAAAATATTTCTTTAAAGCATATTCAAATACCATTATTTCCTTGATCTGTTTTTATTACTACTATTTTATTACTTTTTTCTTTACCTGTATTAGCAGGTGCTATTACTATATTATTATTTGATCGAAATTTTAATACCTCATTTTTTGATCCGACAGGAGGGGGAGATCCTATTCTTTATCAACAT------"
```

Or you can index to a specific sequence like


```r
res$fasta['GBAH0293-06']
```

```
## $`GBAH0293-06`
## [1] "------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------TTAATGTTAGGGATTCCAGATATAGCTTTTCCACGAATAAATAATATTAGATTTTGACTGTTACCTCCATCTTTAATATTATTACTTTTAAGAAATTTTTTAAATCCAAGTCCTGGAACAGGATGAACAGTTTATCCTCCTTTATCATCAAATTTATTTCATTCTTCTCCTTCAGTTGATTTAGCAATTTTTTCTTTACATATTTCAGGTTTATCTTCTATTATAGGTTCATTAAATTTTATTGTTACAATTATTATAATAAAAAATATTTCTTTAAAATATATTCAATTACCTTTATTTTCTTGATCTGTATTTATTACTACTATTCTTTTATTATTTTCTTTACCTGTATTAGCTGGAGCTATTACTATATTATTATTTGATCGAAATTTTAATACATCTTTTTTTGATCCAACAGGAGGGGGAGATCCAATTCTTTATCAACATTTATTTTGATTTTTTGGTCATCCTGAAGTTTATATTTTAATTTTACCTGGATTTGGATTAATTTCTCAAATTATTTCTAATGAAAGAGGAAAAAAAGAAACTTTTGGAAATATTGGTATAATTTATGCTATATTAAGAATTGGACTTTTAGGTTTTATTGTT---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------"
```

### Get trace files

This function downloads files to your machine - it does not load them into your R session - but prints out where the files are for your information.


```r
bold_trace(taxon='Osmia', quiet=TRUE)
```

```
## Trace file extracted with files: 
## 
## /Users/sacmac/github/ropensci/bold/inst/vign/bold_trace_files/BBHYL359-10[LepF1,LepR1]_F.ab1
## /Users/sacmac/github/ropensci/bold/inst/vign/bold_trace_files/BBHYL359-10[LepF1,LepR1]_R.ab1
## /Users/sacmac/github/ropensci/bold/inst/vign/bold_trace_files/HMBCH025-07_F.ab1
## /Users/sacmac/github/ropensci/bold/inst/vign/bold_trace_files/HMBCH025-07_R.ab1
## /Users/sacmac/github/ropensci/bold/inst/vign/bold_trace_files/HMBCH056-07_F.ab1
## /Users/sacmac/github/ropensci/bold/inst/vign/bold_trace_files/HMBCH056-07_R.ab1
## /Users/sacmac/github/ropensci/bold/inst/vign/bold_trace_files/HMBCH063-07_F.ab1
## /Users/sacmac/github/ropensci/bold/inst/vign/bold_trace_files/HMBCH063-07_R.ab1
## /Users/sacmac/github/ropensci/bold/inst/vign/bold_trace_files/HMBCH064-07_F.ab1
## /Users/sacmac/github/ropensci/bold/inst/vign/bold_trace_files/HMBCH064-07_R.ab1
## /Users/sacmac/github/ropensci/bold/inst/vign/bold_trace_files/Osm_aur_T505_LCOHym_D04_008_copy.ab1
## /Users/sacmac/github/ropensci/bold/inst/vign/bold_trace_files/Osm_aur_T505_NancyFull_D10_008_copy.ab1
## /Users/sacmac/github/ropensci/bold/inst/vign/bold_trace_files/Osm_ruf_T309_LCOHym_C06_006_copy.ab1
## /Users/sacmac/github/ropensci/bold/inst/vign/bold_trace_files/Osm_ruf_T309_Nancy_C06_006_copy.ab1
## /Users/sacmac/github/ropensci/bold/inst/vign/bold_trace_files/SSBAC3522-12[LepF1,LepR1]_F.ab1
## /Users/sacmac/github/ropensci/bold/inst/vign/bold_trace_files/SSBAC3522-12[LepF1,LepR1]_R.ab1
## /Users/sacmac/github/ropensci/bold/inst/vign/bold_trace_files/SSWLB1504-13[LepF1,LepR1]_R.ab1
## /Users/sacmac/github/ropensci/bold/inst/vign/bold_trace_files/SSWLB397-13[LepF1,LepR1]_F.ab1
## /Users/sacmac/github/ropensci/bold/inst/vign/bold_trace_files/SSWLB397-13[LepF1,LepR1]_R.ab1
## /Users/sacmac/github/ropensci/bold/inst/vign/bold_trace_files/TRACE_FILE_INFO.txt
```
